File: solutions/clojure/darts/1/src/darts.clj
(ns darts)

(defn score
  "Calculates the score of a dart throw"
  [x y]
  (let [r (Math/sqrt (+ (* x x) (* y y)))]
    (cond
      (<= r 1.0) 10
      (<= r 5.0) 5
      (<= r 10.0) 1
      :else 0)))

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {

    private final int INNER_RADIUS = 1;
    private final int MIDDLE_RADIUS = 5;
    private final int OUTER_RADIUS = 10;

    int score(double xOfDart, double yOfDart) {
        double dartRadius = getDistanceFromCenter(xOfDart, yOfDart);
        int score = 0;
        if (dartRadius <= 1d) {
            score = 10;
        } else if (dartRadius <= 5d) {
            score = 5;
        } else if (dartRadius <= 10d) {
            score = 1;
        }
        return score;
    }

    private double getDistanceFromCenter(double x, double y) {
        return Math.sqrt(x * x + y * y);
    };
}
